kuma/dashboards/views.py

```diff
@@ -98,6 +95,13 @@ def revisions(request):
     if query_kwargs or exclude_kwargs:
         revisions = revisions.filter(**query_kwargs).exclude(**exclude_kwargs)
 
+    # prefetch_related needs to come after all filters have been applied to qs
+    revisions = (revisions.prefetch_related('akismet_submissions',
```

Comment:
`akismet_submissions` is only can be accessed by  the user having the proper permission. most of the user does not need it.
so only prefetch `akismet_submissions` only if `show_spam_submission` is `True

Something like
```python
show_spam_submission = (request.user.is_authenticated() and
    request.user.has_perm('wiki.add_revisionakismetsubmission'))

if show_spam_submission:
    revisions = revisions.prefetch_related('akismet_submissions')

Reply:
Good point, will update. Thanks again!